src/main/webapp/app/forms/form-footer/form-footer.component.html

```diff
@@ -0,0 +1,32 @@
+<div class="update-programming-exercise-footer gap-4">
+    <jhi-exercise-update-notification
+        class="flex-grow-1"
+        [isCreation]="isCreation() ?? false"
+        [isImport]="isImport() ?? false"
+        [notificationText]="notificationText"
+        (notificationTextChange)="notificationTextChange.next($event)"
+    />
+    <div class="update-programming-exercise-footer-content">
+        @if (isSaving) {
+            <span jhiTranslate="artemisApp.quizExercise.edit.saving" class="badge bg-secondary"> </span>
+        }
+        <ng-template #tooltipTranslate>
+            @for (reason of invalidReasons; track reason) {
+                <div>
+                    <p jhiTranslate="{{ reason.translateKey }}"></p>
+                </div>
+            }
+        </ng-template>
+        @if (invalidReasons.length > 0) {
+            <span class="badge bg-danger" [ngbTooltip]="tooltipTranslate" tooltip-placement="top-right auto">
+                <fa-icon [icon]="faExclamationCircle" />
+                <span jhiTranslate="artemisApp.quizExercise.edit.invalidInput"></span>
+                <span>({{ invalidReasons.length }})</span>
+            </span>
+        }
+        <button type="button" id="cancel-save" class="btn btn-secondary me-1" (click)="cancel.next()">
+            <fa-icon [icon]="faBan" />&nbsp;<span jhiTranslate="entity.action.cancel">Cancel</span>
+        </button>
+        <jhi-button id="save-entity" [disabled]="invalidReasons.length > 0 || isDisabled" [isLoading]="isSaving" [icon]="faSave" [title]="saveTitle()" (onClick)="save.next()" />
```

Comment:
Since the title generating relys on singals there is no/not much performance overhead. 
In that way we reduce the code, keep it more maintainable and have the logic inside the ts file.

i would rather keep it like it is.

Reply:
Mhh, the method may depend on the signals but it itself is not a signal. So I'm pretty sure the drawback of using methods in templates still applies (evaluating the value every turn instead of only when it changes).

What do you think?

Reply:
No it is actually a signal because `compute` calcualtes a new Signal with the signals used as dependency, so only evaluated when one of those changes.

https://angular.io/api/core/computed

Reply:
Oh, yeah. I see. I'm learning here. Very good stuff. Thanks.